spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/TestFunctionCatalog.java

```diff
@@ -27,107 +30,106 @@
 import org.apache.spark.sql.connector.catalog.FunctionCatalog;
 import org.apache.spark.sql.connector.catalog.Identifier;
 import org.apache.spark.sql.connector.catalog.functions.UnboundFunction;
-import org.assertj.core.api.Assertions;
-import org.junit.After;
-import org.junit.Assert;
-import org.junit.Before;
-import org.junit.Test;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.TestTemplate;
 
-public class TestFunctionCatalog extends SparkTestBaseWithCatalog {
+public class TestFunctionCatalog extends TestBaseWithCatalog {
   private static final String[] EMPTY_NAMESPACE = new String[] {};
   private static final String[] SYSTEM_NAMESPACE = new String[] {"system"};
   private static final String[] DEFAULT_NAMESPACE = new String[] {"default"};
   private static final String[] DB_NAMESPACE = new String[] {"db"};
-  private final FunctionCatalog asFunctionCatalog;
+  private FunctionCatalog asFunctionCatalog;
 
-  public TestFunctionCatalog() {
+  @BeforeEach
+  public void before() {
+    super.before();
     this.asFunctionCatalog = castToFunctionCatalog(catalogName);
+    createDefaultNamespace();
   }
 
-  @Before
   public void createDefaultNamespace() {
```

Comment:
no need to have this separate method now. Can you just move `sql("CREATE NAMESPACE IF NOT EXISTS %s", catalogName + ".default");` to `before()`?